lib/ProductOpener/Redis.pm

```diff
@@ -0,0 +1,50 @@
+
+package ProductOpener::Redis;
+
+use ProductOpener::PerlStandards;
+use Exporter qw< import >;
+
+BEGIN
+{
+	use vars       qw(@ISA @EXPORT_OK %EXPORT_TAGS);
+	@EXPORT_OK = qw(
+		&push_to_search_service
+
+		);    # symbols to export on request
+	%EXPORT_TAGS = (all => [@EXPORT_OK]);
+}
+
+use vars @EXPORT_OK ;
+
+use Log::Any qw($log);
+use ProductOpener::Config2;
+use Redis::Client;
+
+sub init_redis() {
+	if ($ProductOpener::Config2::redis_url eq "")
+	{
+		$log->warn("Redis URL not provided for search indexing", { error => $@ }) if $log->is_warn();
+		return undef;
+	}
+	eval {
+		my $redis_client = Redis::Client->new(host => $ProductOpener::Config2::redis_url);
+		return $redis_client;
+	
+	};
+	if ($@) {
+		$log->warn("Error connecting to Redis", { error => $@ }) if $log->is_warn();
+	}
+	return undef;
+}
+
+my $redis_client = init_redis();
+
+sub push_to_search_service($product_ref) {
+	if (defined($redis_client)) {
+                $redis_client->rpush('search_import_queue', $product_ref->{code});
```

Comment:
I think we need an eval block here, as otherwise we get an error when redis is not reachable.

Software error:
Can't connect to Redis host at searchredis:6379:6379: IO::Socket::INET: Bad hostname 'searchredis:6379' at /opt/perl/local/lib/perl5/Redis/Client/Role/URP.pm line 21.
For help, please send mail to the webmaster ([contact@openfoodfacts.localhost](mailto:contact@openfoodfacts.localhost)), giving this error message and the time and date of the error.

Reply:
Sure, done!